src/openrct2-win/openrct2-win.cpp

```diff
@@ -57,19 +56,9 @@ static char** GetCommandLineArgs(int argc, wchar_t** argvW)
             argv[i] = ConvertWideChartoUTF8(argvW[i]);
         }
     }
-    return argv;
-}
 
-static void FreeCommandLineArgs(int argc, char** argv)
-{
-    // Free argv
-    for (int i = 0; i < argc; i++)
-    {
-        free(argv[i]);
-    }
-    free(argv);
+    return argv.get();
```

Comment:
Here the code is returning the pointer that is managed by `argv` to the caller. As soon as this function ends, `argv` is destroyed, and the resource is freed.
So in the callsite:
```
char** thePtr = GetCommandLineArgs(...)
```
`thePtr` no longer points to a valid resource, since it was destroyed when `GetCommandLineArgs` ended.

What you want to do here is to return `argv` and make this function return a `std::unique_ptr<char*[]>`. By doing so, the caller will still hold the resource it needs for as long as it needs.

Reply:
Changes made.Please review.